### PR TITLE
Added safetydb datasource

### LINKED ISSUE #1108: Consider safetydb for vulntotal
We have dropped the support for safetydb in vulnerablecode here https://github.com/nexB/vulnerablecode/commit/594c4422c13e9da5527016b04269155770aadb5e, but it might be used in vulntotal

### PR DESCRIPTION
**Pull request by OmkarPh**
Fixes #1108

### CODE REVIEW

File: vulntotal/datasources/safetydb.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            logger.error(f"Error while fetching safetydb advisories: {e}")
+            return
+
+        return response.json()
+
+    def datasource_advisory(self, purl) -> Iterable[VendorData]:
+        if purl.type != "pypi":
```
Comment (keshav-space) [inline vulntotal/datasources/safetydb.py:49]: Use the `supported_ecosystem` to skip the unsupported PURLs.

```suggestion
        if purl.type not in self.supported_ecosystem():
```

**Review by keshav-space — CHANGES_REQUESTED** [review]
Thanks @OmkarPh, LGTM, just a minor nit for your consideration.
Also please rebase your branch.

File: vulntotal/datasources/safetydb.py (hunk 1 above)
Comment (OmkarPh) [reply vulntotal/datasources/safetydb.py]: Thanks keshav, I've rebased & updated

**Review by keshav-space — APPROVED** [review]
Merging!

**Pull request merged**